### PR TITLE
Extract part of Semver compatibility logic to shared package and implement in Invoices and Taxes

### PR DESCRIPTION
**Pull request by lkostrowski**

### CODE REVIEW

File: .changeset/famous-onions-jam.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+"saleor-app-taxes": minor
+---
+
+Set minimum Saleor version where app can be installed (3.10).
```
Comment (lkostrowski) [inline .changeset/famous-onions-jam.md:5]: @peelar I dont know the exact version that app is compatible with. Please confirm, I can change it

Comment (peelar) [inline .changeset/famous-onions-jam.md:5]:
```suggestion
Set minimum Saleor version where app can be installed (3.9).
```

File: apps/taxes/saleor-app.ts [hunk 2]
```diff
@@ -34,3 +34,5 @@ switch (process.env.APL) {
 export const saleorApp = new SaleorApp({
   apl,
 });
+
+export const REQUIRED_SALEOR_VERSION = ">=3.10 <4";
```
Comment (peelar) [inline apps/taxes/saleor-app.ts:38]:
```suggestion
export const REQUIRED_SALEOR_VERSION = ">=3.9 <4";
```

File: packages/shared/src/saleor-version-compatibility-validator.ts [hunk 3]
```diff
@@ -1,8 +1,5 @@
 const semver = require("semver");
 
-/**
- * TODO Extract to shared or even SDK
- */
 export class SaleorVersionCompatibilityValidator {
   constructor(private appRequiredVersion: string) {}
```
Comment (krzysztofwolski) [inline packages/shared/src/saleor-version-compatibility-validator.ts:5]: What do you think about changing the error message to 
`Your Saleor version (${saleor version}) is not compatible with the App (${this.appRequiredVersion})`

Not these words exactly, but it will speed up debugging if the message sent to the User will contain both requirement and his current version. WDYT?

File: apps/taxes/src/pages/api/register.ts [hunk 4]
```diff
@@ -25,4 +37,46 @@ export default createAppRegisterHandler({
...
+        return {
+          token,
+        };
+      });
+
+      const saleorVersion = await client
+        .query<SaleorVersionQuery>(SaleorVersion, {})
+        .toPromise()
+        .then((res) => {
+          return res.data?.shop.version;
```
Comment (krzysztofwolski) [inline apps/taxes/src/pages/api/register.ts:59]: Please check if running Saleor core from the main branch will pass this test. AFAIR they do not have version on main, just use `dev` tag.
If so, should we add edge case for it?

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]

File: packages/shared/src/saleor-version-compatibility-validator.ts (hunk 3 above)
Comment (krzysztofwolski) [reply packages/shared/src/saleor-version-compatibility-validator.ts:5]: Absolute bomb if error would emit "user friendly" message like "Update Saleor version", or "App is not yet compatible with new Saleor version you are using".
But it might require a bit of fun with semver and possible edge cases.

File: apps/taxes/src/pages/api/register.ts (hunk 4 above)
Comment (lkostrowski) [reply apps/taxes/src/pages/api/register.ts:59]: @krzysztofwolski main is on master-staging right? Apps install there properly AFAIK

File: packages/shared/src/saleor-version-compatibility-validator.ts (hunk 3 above)
Comment (lkostrowski) [reply packages/shared/src/saleor-version-compatibility-validator.ts:5]: Improved error message to include both versions

File: apps/taxes/src/pages/api/register.ts (hunk 4 above)
Comment (krzysztofwolski) [reply apps/taxes/src/pages/api/register.ts:59]: Great!

**Review by krzysztofwolski — APPROVED** [review]

**Review by peelar — APPROVED** [review]
